src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointInfoBuilderExtensions.cs

```diff
@@ -0,0 +1,336 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Constraints;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public class MvcEndpointInfo
```

Comment:
Does this need to be public? It seems like this could all be hidden behind the extension methods

Reply:
I don't think it needs to be but if we made it internal then no one else would be able to use the settings from the extension methods. The pattern with options objects appear to be that settings are left open for everyone to see.

Reply:
Oh ah I see, you've got this hanging off of an options type. That's OK for now.